src/irbasis3/poly.py

```diff
@@ -87,6 +88,50 @@ def value(self, l, x):
         res *= self._norm[i]
         return res
 
+    def overlap(self, f, deg=None, axis=None):
+        """
+        Evaluate overlap \int dx p(x) f(x) using piecewise Gauss-Legendre quadrature,
+        where $p(x)$ are the polynomials.
+
+        f: funtion-like object that evalues f(x) for a 1D array of x.
+           The first axis of the result must correspond to x axis.
+
+        deg: None or int, optional
+            Degree of Gauss-Legendre quadrature rule
+            By default, we set deg to 2 * polyorder.
```

Comment:
It would be cool to use adaptive quadrature here, say Gauss-Kronrod [GK 21/10] quadrature rather than having to hard-code the degree here.  But we can do this later...

[GK 21/10]: https://github.com/scipy/scipy/blob/47bb6febaa10658c72962b9615d5d5aa2513fa3a/scipy/integrate/_quad_vec.py#L520]

Reply:
Yes, we can do implement adaptive quadrature later.
From my experience, this hard-code version works very well.